File: iHome.Web/src/app/model/car.ts
import { Color } from "@angular-material-components/color-picker";
import { LoginService } from "../services/LoginService";

export class Car {

    constructor(private loginService: LoginService) {
        this.UserName = this.loginService.getSnapshotCurrentUser().userName;
        this.TechnicalInspection = [];
        this.TechnicalRevision = [];
    }

    ColorHex: string;
    private _color: Color;
    public set Color(value: Color) {
        this._color = value;
        if (this._color)
            this.ColorHex = this._color.hex;
    }
    public get Color() {
        return this._color;
    }

    Id: string;
    TypeName: string;
    CylinderCapacity: number;
    ManufactureDate: Date;
    Brand: string;
    Price: number;
    UserName: string;
    TechnicalInspection: PeriodicTechnicalInspection[];
    TechnicalRevision: PeriodicTechnicalRevision[];
}

export class PeriodicTechnicalInspection {
    Id: string;
    Passed: boolean;
    Kilometers: number;
    Notes: string;
    ServiceGarage: string;
    Price: number;
    Date: Date;
}

export class PeriodicTechnicalRevision {
    Id: string;
    Kilometers: number;
    Notes: string;
    ServiceGarage: string;
    Price: number;
    Date: Date;
}

export interface CarResolved {
    car: Car;
    error?: any;
}

File: iHome.Model/Concrete/Car.cs
using iHome;
using System;
using System.ComponentModel.DataAnnotations;
using System.Collections.Generic;
using iHome.Model;
using System.Drawing;

namespace iHome.Model
{
    public class Car : Record
    {
        private List<PeriodicTechnicalRevision> _revisions;
        private List<PeriodicTechnicalInspection> _inspections;
        public Car()
        {
            _revisions = new List<PeriodicTechnicalRevision>();
            _inspections = new List<PeriodicTechnicalInspection>();
        }

        public string UserName { get; set; }
        public CarType Type { get; set; }
        public string Color { get; set; }
        public double CylinderCapacity { get; set; }
        public DateTime ManufactureDate { get; set; }
        public string Brand { get; set; }
        public double Price { get; set; }
        public List<PeriodicTechnicalRevision> TechnicalRevision
        {
            get => _revisions;
        }
        public List<PeriodicTechnicalInspection> TechnicalInspection
        {
            get => _inspections;
        }
    }
}
